api/app.go

```diff
@@ -39,35 +40,134 @@ func main() {
 }
 
-func (context *httpContext) getPollen(responseWriter http.ResponseWriter, request *http.Request) {
-	output := json.NewEncoder(responseWriter)
-	vars := mux.Vars(request)
-
-	date, err := time.Parse(time.RFC3339, vars["date"])
+func writeObject(responseWriter http.ResponseWriter, output *json.Encoder, object interface{}, err error) {
 	if err != nil {
-		responseWriter.WriteHeader(400)
+		responseWriter.WriteHeader(http.StatusInternalServerError)
 		output.Encode(err)
 		return
 	}
+	if object == nil {
+		responseWriter.WriteHeader(http.StatusNotFound)
+		output.Encode("Object not found")
+		return
+	}
+	output.Encode(object)
+}
+
+func (context *httpContext) getPollenTypes(responseWriter http.ResponseWriter, request *http.Request) {
+	output := json.NewEncoder(responseWriter)
+
+	types := []dataaccess.PollenType{
+		dataaccess.PollenTypeGrass,
+		dataaccess.PollenTypeBirch,
+	}
+
+	result := make([][]interface{}, len(types))
+
+	for i, pollenType := range types {
+		result[i] = []interface{}{pollenType, pollenType.String()}
+	}
 
-	pollenData, err := context.Repo.GetPollen(dataaccess.TimestampToDate(date))
+	output.Encode(result)
+}
+
+func (context *httpContext) getLocation(responseWriter http.ResponseWriter, request *http.Request) {
+	output := json.NewEncoder(responseWriter)
+	vars := mux.Vars(request)
+
+	locationID, err := strconv.Atoi(vars["location"])
 	if err != nil {
-		responseWriter.WriteHeader(500)
+		responseWriter.WriteHeader(http.StatusBadRequest)
 		output.Encode(err)
 		return
 	}
-	if pollenData == nil {
-		responseWriter.WriteHeader(404)
-		output.Encode("Date not found")
+
+	location, err := context.Repo.GetLocation(locationID)
+
+	writeObject(responseWriter, output, location, err)
+}
+
+func (context *httpContext) searchLocation(responseWriter http.ResponseWriter, request *http.Request) {
+	output := json.NewEncoder(responseWriter)
+
+	country := request.FormValue("country")
+	city := request.FormValue("city")
+	if country == "" && city == "" {
+		responseWriter.WriteHeader(http.StatusBadRequest)
 		return
 	}
-	output.Encode(pollenData)
+
+	location, err := context.Repo.SearchLocation(country, city)
+
+	writeObject(responseWriter, output, location, err)
+}
+
+func (context *httpContext) getPollen(responseWriter http.ResponseWriter, request *http.Request) {
+	output := json.NewEncoder(responseWriter)
+	vars := mux.Vars(request)
+
+	var date time.Time
+	if vars["date"] == "tomorrow" {
+		date = time.Now().AddDate(0, 0, 1)
+	} else {
+		var err error
+		date, err = time.Parse(time.RFC3339, vars["date"])
+		if err != nil {
+			responseWriter.WriteHeader(http.StatusBadRequest)
+			output.Encode(err)
+			return
+		}
+	}
+
+	// Parse pollen type
+	pollenType, err := strconv.Atoi(request.FormValue("pollentype"))
+	if err != nil {
+		// TODO: return error when obsolete is removed entirely
+		responseWriter.Header().Set("X-Obsolete-pollentype", "Calling this endpoint without declaring pollentype in query is obsolete")
```

Comment:
When is this scheduled for deletion? Is it when [Tomorrow's Pollen Today](https://github.com/Tomorrows-pollen-today/tomorrows-pollen-today) gets updated to use the new format?

Reply:
Correct. The function docstring now includes this detail.